src/interface/extension_interface.hpp

```diff
@@ -139,6 +139,101 @@ _matcopy_impl(sb_handle_t& sb_handle, index_t m, index_t n, element_t alpha,
   return ret;
 }
 
+/*!
+ * @brief Wrapper around Transpose-Add. Creates the views, then makes and
+ * launches Transpose Add kernel
+ */
+template <bool both_trans, int Tile_size, int wg_size, int cl_size,
+          bool local_memory, typename sb_handle_t, typename container_0_t,
+          typename container_1_t, typename container_2_t, typename element_t,
+          typename index_t>
+typename sb_handle_t::event_t _transpose_add_impl(
+    sb_handle_t& sb_handle, index_t _M, index_t _N, element_t _alpha,
+    container_0_t a_, index_t _lda, index_t _nrows_a, index_t _ncols_a,
+    element_t _beta, container_1_t b_, index_t _ldb, index_t _nrows_b,
+    index_t _ncols_b, container_2_t c_, index_t _ldc) {
+  constexpr const index_t num_line_elems =
+      std::max(Tile_size, static_cast<int>(cl_size / sizeof(element_t)));
+  constexpr const index_t num_tiles_per_line = num_line_elems / Tile_size;
+  // Matrix Views
+  auto A_view =
+      make_matrix_view<col_major>(a_, _nrows_a, _ncols_a, _lda, (index_t)1);
+  auto B_view =
+      make_matrix_view<col_major>(b_, _nrows_b, _ncols_b, _ldb, (index_t)1);
+
+  auto C_view = make_matrix_view<col_major>(c_, _M, _N, _ldc, (index_t)1);
+
+  // Work items & groups sizes
+  index_t n_wg = ((_M - 1) / Tile_size + 1) * ((_N - 1) / Tile_size + 1);
+  index_t global_size = n_wg * wg_size;
+
+  // Transpose Add expression Tree
+  auto trans_scale_tree =
+      make_transpose_add<both_trans, Tile_size, wg_size, cl_size, local_memory>(
+          A_view, B_view, C_view, _alpha, _beta);
+
+  if constexpr (local_memory) {
+    index_t local_mem = static_cast<index_t>((num_line_elems + 1) * Tile_size /
+                                             num_tiles_per_line);
+    return sb_handle.execute(trans_scale_tree, wg_size, global_size, local_mem);
```

Comment:
```suggestion
    return sb_handle.execute(trans_scale_tree, static_cast<index_t>(wg_size), global_size, local_mem);
```
Otherwise, it gives error for `index_t` different from `int`.